spec/components/topic_query_spec.rb

```diff
@@ -258,11 +259,27 @@
 
       topic_ids = topic_query.list_latest.topics.map(&:id)
       expect(topic_ids).not_to include(muted_topic.id)
+      expect(topic_ids).not_to include(muted_tagged_topic.id)
       expect(topic_ids).to include(tagged_topic.id)
       expect(topic_ids).to include(untagged_topic.id)
 
       topic_ids = topic_query.list_new.topics.map(&:id)
       expect(topic_ids).not_to include(muted_topic.id)
+      expect(topic_ids).not_to include(muted_tagged_topic.id)
+      expect(topic_ids).to include(tagged_topic.id)
+      expect(topic_ids).to include(untagged_topic.id)
+
+      SiteSetting.mute_other_present_tags = false
+
+      topic_ids = topic_query.list_latest.topics.map(&:id)
+      expect(topic_ids).not_to include(muted_topic.id)
+      expect(topic_ids).to include(muted_tagged_topic.id)
+      expect(topic_ids).to include(tagged_topic.id)
+      expect(topic_ids).to include(untagged_topic.id)
```

Comment:
There is a `contain_exactly` matcher in RSpec that you can use to remove the duplication here.

```
expect(topic_ids).to contain_exactly(muted_tag_topic.id, tagged_topic.id, untagged_topic.id)
```